src/main/java/seedu/duke/model/template/BooleanTemplate.java

```diff
@@ -138,9 +144,11 @@ private String getInputEquation(int index) throws DukeException {
 
         // Odd index - first input, even index, 2nd input
         if (index % 2 == 1) {
-            equation += gate.isSetInput() ? gate.getInput() : "?";
+            String input = gate.getInput() ? "1" : "0";
+            equation += gate.isSetInput() ? input : "?";
         } else {
-            equation += gate.isSetSecondInput() ? gate.getSecondInput() : "?";
+            String input = gate.getInput() ? "1" : "0";
+            equation += gate.isSetSecondInput() ? input : "?";
```

Comment:
` String input = gate.getInput() ? "1" : "0";` can be extracted out and placed before the if-else block?

Reply:
😮You spotted a bug!

Reply:
omg hahahaha unbeknownst to me